src/ui/public/agg_types/metrics/derivative.js

```diff
@@ -0,0 +1,142 @@
+import AggTypesMetricsMetricAggTypeProvider from 'ui/agg_types/metrics/metric_agg_type';
+import orderAggTemplate from 'ui/agg_types/controls/sub_agg.html';
+import _ from 'lodash';
+import $ from 'jquery';
+import VisAggConfigProvider from 'ui/vis/agg_config';
+import VisSchemasProvider from 'ui/vis/schemas';
+
+export default function AggTypeMetricDerivativeProvider(Private) {
+  const DerivativeAggType = Private(AggTypesMetricsMetricAggTypeProvider);
+  const AggConfig = Private(VisAggConfigProvider);
+  const Schemas = Private(VisSchemasProvider);
+
+  const aggFilter = ['!top_hits', '!percentiles', '!percentile_ranks', '!median', '!std_dev'];
+  const orderAggSchema = (new Schemas([
+    {
+      group: 'none',
+      name: 'orderAgg',
+      title: 'Order Agg',
+      aggFilter: aggFilter
+    }
+  ])).all[0];
+
+  return new DerivativeAggType({
+    name: 'derivative',
+    title: 'Derivative',
+    makeLabel: function (aggConfig) {
```

Comment:
Looks like this function could be pulled out into a separate file and then tested separately?